### PR TITLE
[Nataliya Mykytyuk] Main header

### PR DESCRIPTION
**Pull request by dnistrova**

### CODE REVIEW

File: src/app/core/header/header-contacts/header-contacts.component.scss [hunk 1]
```diff
@@ -91,4 +88,32 @@ hr {
   font-size: 12px;
   text-align: center;
   display: none;
-}
\ No newline at end of file
+}
+
+@media only screen and (max-width: 767px) {
```
Comment (IgorStetsiuk) [inline src/app/core/header/header-contacts/header-contacts.component.scss]: I would suggest to move all media queries to some shared folder with constants and create `mixins` for it 

Please refer: https://css-tricks.com/snippets/sass/mixin-manage-breakpoints/
(https://responsivedesign.is/develop/getting-started-with-sass-and-breakpoint-mixin/) --optional

They should be reusable among all components and be the same (defined)
These changes concerns all media queries in your `scss` files

File: src/app/core/header/header-nav/header-nav.component.scss [hunk 2]
```diff
@@ -50,15 +62,19 @@
   padding-top: 13px;
   display: flex;
   position: relative;
+  flex-wrap: nowrap;
+  align-items: baseline;
+
   i {
     font-size: 20px;
```
Comment (IgorStetsiuk) [inline src/app/core/header/header-nav/header-nav.component.scss]: Please move it to constants, `font-size`s should be reusable

Please take a look here : https://sass-lang.com/styleguide/typography

#### Revice all of your changes regarding `font-size ` property

File: src/app/core/header/header-nav/header-nav.component.scss [hunk 3]
```diff
@@ -75,5 +91,113 @@
...
+
+  .logo {
+    width: 40%;
+    margin-right: 0;
+    margin-bottom: 0;
+  }
+}
+
+@media only screen and (max-width: 424px) {
+  nav.revealMenu {
```
Comment (IgorStetsiuk) [inline src/app/core/header/header-nav/header-nav.component.scss]: What is a name convention for  `revealMenu`, please be consistent. What name convention you are decided to use (BEM, modules, ets.)

File: src/app/core/header/header-nav/header-nav.component.scss [hunk 4]
```diff
@@ -75,5 +91,113 @@
...
+
+  .main-nav {
+    flex-direction: column;
+    position: absolute;
+    top: 160px;
+    width: 100%;
+  }
+
+  .header-right-bar {
+    padding-bottom: 10px;
```
Comment (IgorStetsiuk) [inline src/app/core/header/header-nav/header-nav.component.scss]: If this `10px` you are using frequently please consider to move it to constants and choose the appropriate name for such variables 

**The same changes concerns other CSS properties**

File: src/app/core/header/header-nav/header-nav.component.scss [hunk 5]
```diff
@@ -75,5 +91,113 @@
...
+    display: none;
+  }
+
+ .main-nav {
+  position: absolute;
+  top: 194px;
+  left: 0;
+ }
+
+  nav.revealMenu {
```
Comment (IgorStetsiuk) [inline src/app/core/header/header-nav/header-nav.component.scss]: As for me it is a not correct `revealMenu` class name. We don't use CamelCase while naming `css` classes, don't we ?

File: src/app/core/header/header-contacts/header-contacts.component.scss [hunk 6]
```diff
@@ -91,4 +88,32 @@ hr {
...
+    align-items: center;
+    .mail {
+      margin-right: 0;
+    }
+  }
+
+  .social-media {
+    width: 100%;
+    text-align: center;
+    a:first-child {
```
Comment (IgorStetsiuk) [inline src/app/core/header/header-contacts/header-contacts.component.scss]: space between rules

File: src/app/core/header/header-contacts/header-contacts.component.scss [hunk 7]
```diff
@@ -91,4 +88,32 @@ hr {
...
+
+  .header-left-bar {
+    flex-direction: column;
+  }
+}
+
+@media only screen and (max-width: 424px) {
+  .header-left-bar {
+    width: 100%;
+    align-items: center;
```
Comment (IgorStetsiuk) [inline src/app/core/header/header-contacts/header-contacts.component.scss]: space between rules

**Comment by IgorStetsiuk** [comment]
We need to move this PR to our main repository prior to Wednesday, please resolve the comment here

File: src/app/core/header/header-contacts/header-contacts.component.scss [hunk 8]
```diff
@@ -35,15 +36,15 @@ i, a {
 .mail {
   margin-right: 15px;
   color: $black-color;
-  font-size: 12px;
+  @include font-size-small;
```
Comment (IgorStetsiuk) [inline src/app/core/header/header-contacts/header-contacts.component.scss]: Nope. It should be as a `variable` not `mixin`. 
Look 
```scss
$gray-some-color-name: #f8f9fa;

`somewhere into file`

font-size: $gray-some-color-name;


```

File: src/app/core/header/header-contacts/header-contacts.component.scss [hunk 9]
```diff
@@ -35,15 +36,15 @@ i, a {
...
   img {
     margin-right: 9px;
   }
 }
 
 .pnone-number {
   color: $black-color;
-  font-size: 12px;
+  @include font-size-small;
```
Comment (IgorStetsiuk) [inline src/app/core/header/header-contacts/header-contacts.component.scss]: variable here not mixin, we are using mixins for example for media queries

File: src/sass/_mixins.scss [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+          justify-content: $justify;
+            -ms-flex-pack: $justify;
+}
+
+@mixin align-items($align) {
+  -webkit-align-items: $align;
+     -moz-align-items: $align;
+      -ms-align-items: $align;
+          align-items: $align;
+}
```
Comment (IgorStetsiuk) [inline src/sass/_mixins.scss]: Please add here `mixins ` for media queries

File: src/sass/media/_mobileL.scss [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+    position: relative;
+  }
+  .main-nav {
+    @include flex-direction(column);
+    position: absolute;
+    top: 160px;
+    width: 100%;
+  }
+  .header-right-bar {
+    padding-bottom: $standart-bottom-padding;
```
Comment (IgorStetsiuk) [inline src/sass/media/_mobileL.scss]: Yes, it is a correct usage of scss capabilities  (variables) -- this should be the same for `font-size`

File: src/sass/media/_mobileL.scss [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+      i {
+        padding-left: 0;
+      }
+    }
+  }
+  .logo {
+    @include flexbox();
+    padding-bottom: $standart-bottom-padding;
+    width: 100%;
+    a {
```
Comment (IgorStetsiuk) [inline src/sass/media/_mobileL.scss]: Look, space between the rules

``` css

.logo {
    @include flexbox();
    padding-bottom: $standart-bottom-padding;
    width: 100%;
   (one space here)
   
    a {
        margin: 0 auto;
    }
```

File: src/app/core/header/header-contacts/header-contacts.component.scss (hunk 8 above)
Comment (IgorStetsiuk) [reply src/app/core/header/header-contacts/header-contacts.component.scss]: Please resolve

File: src/app/app-routing.module.ts [hunk 13]
```diff
@@ -1,7 +1,16 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
+import { PageNotFoundComponent } from './page-not-found/page-not-found.component';
 
-const routes: Routes = [];
+// import routes from './routes';
```
Comment (dverbovyi) [inline src/app/app-routing.module.ts]: solve this

File: src/app/app.component.html [hunk 14]
```diff
@@ -1,3 +1,4 @@
 <app-header></app-header>
-  <router-outlet></router-outlet>
+<!-- <app-content></app-content> -->
```
Comment (dverbovyi) [inline src/app/app.component.html]: ?

File: src/app/app.component.spec.ts [hunk 15]
```diff
@@ -16,3 +16,18 @@ describe('AppComponent', () => {
     }).compileComponents();
   }));
 });
+
+// describe('AppComponent', () => {
```
Comment (dverbovyi) [inline src/app/app.component.spec.ts]: why it's commented?

File: src/app/core/components/content/content.component.html [hunk 16]
```diff
@@ -0,0 +1,3 @@
+<section class="main">
+  <!-- <router-outlet></router-outlet> -->
```
Comment (dverbovyi) [inline src/app/core/components/content/content.component.html]: why is it commented?

File: src/scss/_mixins.scss [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+
+@mixin flex-direction($direction) {
+  -webkit-flex-direction: $direction;
+     -moz-flex-direction: $direction;
+      -ms-flex-direction: $direction;
+          flex-direction: $direction;
+}
+
+@mixin flex-wrap($wrap) {
+  -webkit-flex-wrap: $wrap;
```
Comment (dverbovyi) [inline src/scss/_mixins.scss]: @IgorStetsiuk sass should support autoprefix out of box for properties or via some plugin, isn't?

Comment (IgorStetsiuk) [reply src/scss/_mixins.scss]: Yes, I thought the same way. @dnistrova  Can you please check the same properties into compiled css? Whether it possible

File: src/app/core/components/header/header-nav/header-nav.component.ts [hunk 18]
```diff
@@ -6,6 +6,11 @@ import { Component, OnInit } from '@angular/core';
   styleUrls: ['./header-nav.component.scss']
 })
 export class HeaderNavComponent implements OnInit {
+  searchIsOpen = false;
+
+  clickHandler = () => {
```
Comment (dverbovyi) [inline src/app/core/components/header/header-nav/header-nav.component.ts]: it should be a class method, not a property

onSearchBtnClick() {
    this.isSearchOpen = !this.isSearchOpen
}

File: src/app/core/components/header/header-nav/header-nav.component.ts [hunk 19]
```diff
@@ -6,6 +6,11 @@ import { Component, OnInit } from '@angular/core';
...
+    this.searchIsOpen = !this.searchIsOpen;
```
Comment (dverbovyi) [inline src/app/core/components/header/header-nav/header-nav.component.ts]: rename to isSearchOpen

**Pull request merged**